### ISSUE TITLE
How to setup pre-commit to check diffs of branch for Gitlab?

### ISSUE THREAD

**Issue by voronind** [pre-commit/pre-commit#2011]
Hello. I want to check branch changes only in Gitlab CI.  
I need to use
`pre-commit run --from-ref=? --to-ref=HEAD`
What `--from-ref` to use?

**Comment by asottile** [comment]
please [consult the documentation for running in CI](https://pre-commit.com/#usage-in-continuous-integration)

**Comment by voronind** [comment]
Example from pre-commit documentation raises error.
```console
$ pre-commit run --from-ref=origin/HEAD --to-ref HEAD
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/git', 'diff', '--name-only', '--no-ext-diff', '-z', 'origin/HEAD...HEAD')
return code: 128
expected return code: 0
stdout: (none)
stderr:
    fatal: ambiguous argument 'origin/HEAD...HEAD': unknown revision or path not in the working tree.
    Use '--' to separate paths from revisions, like this:
    'git <command> [<revision>...] -- [<file>...]'
```

**Comment by asottile** [comment]
then you're missing one or more of those revisions -- you may need to `git fetch origin`

**Comment by kevbroch** [comment]
@voronind : Were you able to find a solution?

I was having the same issue.  I looked at the branch the gitlab-runner has for the merge request and there is no origin/HEAD ref.  When I checked out the branch out on my own it is there and the above command works fine.  :shrug: 

My current solution is to add `pre-commit run --from-ref origin/${CI_DEFAULT_BRANCH} --to-ref HEAD` to my .gitlab-ci.yml file
